### PR TITLE
feat: Search Menu Auto Completion

### PR DESCRIPTION
**Pull request by ChristianKienle**
@khmmrl  can you take another look? I have also updated the documentation and the search input now supports two different "confirmation modes": confirm on tab and confirm on enter.

### CODE REVIEW

**Comment by ChristianKienle** [comment]
@CodesOfRa ping :D

File: package.json [hunk 1]
```diff
@@ -1,98 +1,103 @@
...
     "release": "./scripts/publish-release.sh",
     "release:create": "create-release",
     "std-version": "standard-version -m \"chore(release): version %s build ${TRAVIS_BUILD_NUMBER} [ci skip]\"",
-    "test": "vue-cli-service test:unit",
-    "test:watch": "vue-cli-service test:unit --watch",
-    "deploy:docs": "yarn build:docs  && gh-pages -d dist"
+    "test:e2e": "cross-env FD_E2E=true vue-cli-service test:e2e --url http://localhost:10000",
+    "test:watch": "vue-cli-service test:unit --watch"
   },
-  "description": "SAP Fiori Fundamentals, implemented in Vue.js",
```
Comment (CodesOfRa) [inline package.json]: we don't need this anymore?

File: vue.config.js [hunk 2]
```diff
@@ -30,10 +32,10 @@ module.exports = {
...
       })
       .end();
-    config
-      .entry("app")
-      .clear()
-      .add("./src/docs/index.js");
+    // config
+    //   .entry("app")
+    //   .clear()
+    //   .add(isE2e ? "./demo/index.js" : "./src/docs/index.js");
```
Comment (CodesOfRa) [inline vue.config.js]: do we need this? :D maybe we should delete the commented code. what do you think ?

File: package.json (hunk 1 above)
Comment (ChristianKienle) [reply package.json]: this is now in line 4 :)

File: vue.config.js (hunk 2 above)
Comment (ChristianKienle) [reply vue.config.js]: YES. :D Thanks for catching that.

**Comment by khmmrl** [comment]
> @khmmrl  can you take another look? I have also updated the documentation and the search input now supports two different "confirmation modes": confirm on tab and confirm on enter.

As discussed, I would not offer the two options (tab, enter) but always take over the proposal value directly in to the input field when moving the arrow keys. And then enter to confirm/trigger search.

**Comment by ChristianKienle** [comment]
> > @khmmrl  can you take another look? I have also updated the documentation and the search input now supports two different "confirmation modes": confirm on tab and confirm on enter.
> 
> As discussed, I would not offer the two options (tab, enter) but always take over the proposal value directly in to the input field when moving the arrow keys. And then enter to confirm/trigger search.

Yeah. This is still on my list and the last remaining thing from my point of view. But you would still need to react on enter/tab because at some point you want to close the list of suggestions - right?

**Comment by khmmrl** [comment]
> Yeah. This is still on my list and the last remaining thing from my point of view. But you would still need to react on enter/tab because at some point you want to close the list of suggestions - right?

I would say, in this case it would be as follows:
- TAB pressed -> close suggestions + focus on "search button"
- Enter pressed -> close suggestions and trigger search; focus on search results?

**Pull request closed without merging**